### ISSUE TITLE
LOR imports not displaying correctly

### ISSUE THREAD

**Issue by DanJacobsen** [smeighan/xLights#695]
I imported 12 of 16 minitrees from last year's xlights sequences--while also converting them from AC (single channel on/off) to RGB (5 strings of 10 nodes each).   Even though I have render mode set to "erase" now (I used to have it on canvas, but that's about 6 mths ago...), strand #2 seems to have kept its old on/off effects from LOR on a couple of the minitrees.   I only saw this on a sequence where I did some of the original sequence on the AC minitrees in LOR (for Christmas 2015) and imported the effects into xlights.   The "fix" I implemented was to open the strands of the offending minitrees and delete all the specific effects on strand #2, which was always the case on the ones that did not display correctly.  So, the main change this year is I told xlights these are now RGB minitrees and I am now applying RGB effects to RGB models that were originally single channel AC models.   This might not be a GitHub subject, but figured I would post here just in case...

**Comment by AzGilrock** [comment]
So what did you expect to happen?  Should we automatically erase effects when you change string types?

**Comment by DanJacobsen** [comment]
Gil:
I was thinking that when I had erase as the render mode, old effects (in this example what used to be white on/off effects) would be deleted and replaced with the xlights effects when I rendered throughout the sequence.   Perhaps this is just my misunderstanding of erase/canvas modes...    In the specific instances, I replaced white on/off effects (from last year) with either blue on/off or blue/white spirals) Interestingly, even after I deleted some of the blue on/off or spiral effects and placed an OFF effect on the model in that time slot, the old white ON effect remained ON (just on strand #2 for some reason--maybe this is an RGB version of a single channel?) even though the OFF effect was placed on the model.   This will be the only time I will ever deal with this (these are the only AC channels I need to convert to RGB...), which is why I said in the GitHub case that it might not be a really good GitHub topic!!   But I thought it might be valid since others import LOR effects.

Thanks Gil.


Dan E. Jacobsen
(225) 439-4284


________________________________

**Comment by AzGilrock** [comment]
We never erase effects.  Only you can do that.  Erase mode just clears out the render buffer.  You should never need an Off effect except the rare case where you want to blank out an effect from a group.  If data still appears then you either didn't save or render or you didn't drill down to notice you have effects at the strand or node level or you have effects on another model with overlapping channels.  The problem is going to be in your sequence or some step you aren't doing.

**Comment by DanJacobsen** [comment]
Thanks for the clarification.    My misunderstanding on what the erase mode does for you....   I thought, for example, if I changed an all white ON effect to a blue ON effect then rendered/saved, that the old white ON effect was "erased" when in erase mode.   My bad...    I believe my issue here is related to converting an LOR sequence AND converting THOSE models from single channel models to RGB models.   I am 100% that I sequenced the stuff I am messing with in this example in LOR, then imported it as a (single channels) layer in last year's version of xlights.  When I was re-doing these effects with this year's version of xlights, I did not drill down and look for remaining effects for sure!!   I think I said this last night; I went from left to right in the sequence and deleted all occurrences of "stray" effects on strand #2 on the offending models.   All is well now; all stuff that I wanted gone is GONE.   THANKS GIL!


Dan E. Jacobsen
(225) 439-4284


________________________________
